From 5d126fec3723bfb88431ae16addfc4cc34f72e10 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 03 Dec 2023 10:06:41 +0200
Subject: [PATCH] [SW-166459] gaudi3: print cpu pll freq

CPU PLL (PCI pll div0) is configured by HW depending on the efuse
values.
In order to prevent confusion during gaudi3 PO because of wrong efuse
configuration, CPU PLL frequency is printed to UART, and if the
configuration differs from nominal an additional warning is printed.

Change-Id: I6cda70df5ff42b8ba85a3971ab20a2630a4e1750
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 239789e..86936c2 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1377,3 +1377,21 @@
 
 	security_rr_open_pcie_config();
 }
+
+#define CPU_PLL_NOMINAL_FREQ	1000000000
+
+__weak void gaudi3_print_cpu_pll(void)
+{
+	u32 cpu_pll_freq;
+
+	cpu_pll_freq = hl_pll_get_freq(mmD0_G4_PCI_PLL_CTRL_BASE + mmPLL_CTRL_CFG, HL_PLL_DIV0);
+	preboot_puts_hex("CPU PLL Frequency = ", cpu_pll_freq, false);
+
+	if (cpu_pll_freq != CPU_PLL_NOMINAL_FREQ)
+		preboot_puts("Warning: CPU PLL freq differs from nominal (1GHz)\n");
+}
+
+void ppboot_plls_init(void)
+{
+	gaudi3_print_cpu_pll();
+}
diff --git a/arch/gaudi3/arc/hl.h b/arch/gaudi3/arc/hl.h
index e676700..1a714f5 100644
--- a/arch/gaudi3/arc/hl.h
+++ b/arch/gaudi3/arc/hl.h
@@ -52,5 +52,6 @@
 int gaudi3_hbw_memcpy(u64 src, u64 dst, u32 size);
 void pif_arc_init_pcie(void);
 void hl_pcie_clear_boot_termination(void);
+void gaudi3_print_cpu_pll(void);
 
 #endif /* ARCH_GAUDI3_ARC_HL_H_ */
diff --git a/arch/gaudi3/arc/onix.c b/arch/gaudi3/arc/onix.c
index bbb4fc3..69af77d 100644
--- a/arch/gaudi3/arc/onix.c
+++ b/arch/gaudi3/arc/onix.c
@@ -74,3 +74,6 @@
 {
 	return PCI_IDS_GAUDI3_FPGA;
 }
+
+/* plls are not implemented on onix platform */
+void gaudi3_print_cpu_pll(void) {}
